### ISSUE TITLE
Stray jobs reported with private/public packit mixture

### ISSUE THREAD

**Issue by lukaszachy** [packit/packit-service#1630]
Reported jobs for  https://github.com/teemtee/tmt/pull/1463 seems a bit off:
You can see that "testing-farm:fedora-36-x86_64:internal-plugins — idorax can't run tests internally" 
however for "testing-farm:centos-stream-8-x86_64:internal-plugins - Test passed..."

When I opened "Details" I saw that all testing-farm:centos-stream-8-x86_64 jobs were pointing to the same Dashboard / Testing farm link. So in the end packit ran only "testing-farm:centos-stream-8-x86_64" but reported it as a success for "internal-plugins" and "internal-wow" as well. 

Currently .packit is set to execute only fedora-latest-stable for internal jobs.

Tl;dr: More jobs are reported than executed.

![image](https://user-images.githubusercontent.com/1548503/186366379-2c8c9b47-a87e-44b5-9808-bd5777ec582a.png)

PR is under development so jobs will change... But IIRC stray jobs appear after some time (fresh results are for just correct jobs, few hours later the stray centos jobs appear).

**Comment by lbarcziova** [comment]
Thanks for the report! I checked the PR pushes and this doesn't happen all the time and happens randomly for targets (I saw this happened for some different push for `fedora-rawhide`), right?

I am wondering whether this could be happening if the statuses are updated during the test babysitting job (it [looks like](https://github.com/packit/packit-service/blob/9ba9e1fe5a4a0952724bca7b0521418dabdddf49/packit_service/worker/helpers/build/babysit.py#L113) we don't run `pre_check` that checks the identifiers equality when called in babysitting job).

We need to investigate this more and hopefully will fix this soon.

**Comment by lbarcziova** [comment]
Ok, in the end, I found the logs when this is happening and it looks like this is really coming from the babysit task:
```
[2022-08-23 13:53:29,901: DEBUG/ForkPoolWorker-1] Checking status of TF pipeline 8f68b8c5-f60a-4626-b467-56027e1a73b8
[2022-08-23 13:53:30,395: DEBUG/ForkPoolWorker-1] Result for the TF pipeline 8f68b8c5-f60a-4626-b467-56027e1a73b8 is passed.
```

```
[2022-08-23 13:53:31,667: DEBUG/ForkPoolWorker-1] Testing farm 8f68b8c5-f60a-4626-b467-56027e1a73b8 result:
passed
[2022-08-23 13:53:31,726: DEBUG/ForkPoolWorker-1] Status reporter will report for GithubProject(namespace="teemtee", repo="tmt"), commit=f15b7000b121ce0b48e372c38db0e06b2628acfe, pr=1463
[2022-08-23 13:53:31,727: DEBUG/ForkPoolWorker-1] Setting Github status check 'success' for check 'testing-farm:centos-stream-9-x86_64': Tests passed ...
[2022-08-23 13:53:32,518: DEBUG/ForkPoolWorker-1] Removing contents of the PV.
[2022-08-23 13:53:32,518: DEBUG/ForkPoolWorker-1] Testing farm 8f68b8c5-f60a-4626-b467-56027e1a73b8 result:
passed
[2022-08-23 13:53:32,625: DEBUG/ForkPoolWorker-1] Status reporter will report for GithubProject(namespace="teemtee", repo="tmt"), commit=f15b7000b121ce0b48e372c38db0e06b2628acfe, pr=1463
[2022-08-23 13:53:32,626: DEBUG/ForkPoolWorker-1] Setting Github status check 'success' for check 'testing-farm:centos-stream-9-x86_64:internal-plugins': Tests passed ...
[2022-08-23 13:53:33,415: DEBUG/ForkPoolWorker-1] Removing contents of the PV.
[2022-08-23 13:53:33,416: DEBUG/ForkPoolWorker-1] Testing farm 8f68b8c5-f60a-4626-b467-56027e1a73b8 result:
passed
[2022-08-23 13:53:33,525: DEBUG/ForkPoolWorker-1] Status reporter will report for GithubProject(namespace="teemtee", repo="tmt"), commit=f15b7000b121ce0b48e372c38db0e06b2628acfe, pr=1463
[2022-08-23 13:53:33,526: DEBUG/ForkPoolWorker-1] Setting Github status check 'success' for check 'testing-farm:centos-stream-9-x86_64:internal-wow': Tests passed ...
[2022-08-23 13:53:34,303: DEBUG/ForkPoolWorker-1] Checking status of TF pipeline 09308ba6-e7f9-4900-a39c-49b4e3754ce0
```

**Comment by lbarcziova** [comment]
The fix should be in production next Tuesday.
